extensions/spring-di/deployment/src/main/java/io/quarkus/spring/di/deployment/SpringDIProcessor.java

```diff
@@ -265,6 +265,14 @@ private void visitAnnotation(final DotName clazz, final IndexView index, final M
             final Set<DotName> scopeStereotypes = new HashSet<>();
             final Set<String> names = new HashSet<>();
             final Set<DotName> clazzAnnotations = classInfo.annotations().keySet();
+
+            for (AnnotationInstance instance : classInfo.classAnnotations()) {
+                // make sure that we don't mix and match Spring and CDI annotations since this can cause a lot of problems
+                if (BuiltinScope.from(instance.name()) != null) {
```

Comment:
Hm, the only problem here is that the set of scopes is extensible. But I don't think that's something we can fix with the current API. I've created https://github.com/quarkusio/quarkus/issues/2188.

Reply:
:+1: